libsodium-sys/src/crypto_generichash.rs

```diff
@@ -25,6 +28,27 @@ extern {
         key: *const u8,
         keylen: size_t)
         -> c_int;
+
+    pub fn crypto_generichash_init(
+        state: *mut crypto_generichash_state,
+        key: *const u8,
+        keylen: size_t,
+        outlen: size_t)
+        -> c_int;
+
+    pub fn crypto_generichash_update(
+        state: *mut crypto_generichash_state,
+        in_: *const u8,
+        inlen: c_ulonglong)
+        -> c_int;
+
+    pub fn crypto_generichash_final(
+        state: *mut crypto_generichash_state,
+        out: *mut u8,
+        outlen: size_t)
+        -> c_int;
+
+    pub fn crypto_generichash_statebytes() -> size_t;
```

Comment:
Also add a constant `CRYPTO_GENERICHASH_STATEBYTES` and a test to see that it's equal to the value this function returns.